File: csharp/prime-factors/PrimeFactors.cs
using System;
using System.Collections.Generic;

public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        long testVal = 2L;
        var factors = new List<long>();
        while (testVal <= number) {
            if (number % testVal == 0) {
                factors.Add(testVal);
                number /= testVal;
            } else {
                testVal++;
            }
        }
        return factors.ToArray();
    }
}

File: dart/prime-factors/lib/prime_factors.dart
class PrimeFactors {
  List<int> factors(int n) {
    List<int> primeFactors = [];

    for (int factor = 2; n >= factor; factor++) {
      while (n % factor == 0) {
        primeFactors.add(factor);
        n ~/= factor;
      }
    }
    return primeFactors;
  }
}
